com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_twoStage

https://hazelcast-l337.ci.cloudbees.com/view/Jet/job/jet/lastCompletedBuild/com.hazelcast.jet$hazelcast-jet-core/testReport/com.hazelcast.jet.core/JobRestartWithSnapshotTest/when_nodeDown_then_jobRestartsFromSnapshot_twoStage/

```
Error Message
Snapshots map not empty after job finished
Stacktrace
java.lang.AssertionError: Snapshots map not empty after job finished
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot(JobRestartWithSnapshotTest.java:267)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_twoStage(JobRestartWithSnapshotTest.java:117)
```

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java
@@ -552,10 +552,13 @@ public class MasterContext {
             logger.warning(String.format("Execution of %s failed after %,d ms", jobIdString(), elapsed), failure);
         }
 
+        JobStatus status = isSuccess(failure) ? COMPLETED : FAILED;
+        jobStatus.set(status);
+
         try {
             coordinationService.completeJob(this, executionId, System.currentTimeMillis(), failure);
         } catch (RuntimeException e) {
-            logger.warning("Completion of " + jobIdString() + " failed", failure);
+            logger.warning("Completion of " + jobIdString() + " failed", e);
         } finally {
             setFinalResult(failure);
         }
@@ -594,8 +597,6 @@ public class MasterContext {
     }
 
     void setFinalResult(Throwable failure) {
-        JobStatus status = isSuccess(failure) ? COMPLETED : FAILED;
-        jobStatus.set(status);
         if (failure == null) {
             completionFuture.internalComplete();
         } else {

